### PR TITLE
Sklearn: validation set weights

### PR DESCRIPTION
**Pull request by pdavalo**
In Sklearn wrapper, the option to train using using instance weights works OK. However, when evaluating performance metrics on validation sets, there was no option to use instance weights.

### CODE REVIEW

**Comment by pdavalo** [comment]
Note that this PR solves the previously posted issue https://github.com/dmlc/xgboost/issues/1804

**Comment by tqchen** [comment]
requires rebase with master, feel free to reopen so it triggers a new PR
